src/audioVisualizer.ts

```
@@ -80,68 +80,184 @@ export class AudioVisualizer {
+   */
+  private sanitizeLogText(text: string): string {
+    if (!text) return '';
+
+    // Replace newlines and carriage returns with spaces
+    let sanitized = text.replace(/[\r\n]+/g, ' ');
+
+    // Collapse multiple spaces into one
+    sanitized = sanitized.replace(/\s+/g, ' ');
+
+    // Remove ANSI escape codes
+    sanitized = sanitized.replace(/\x1b\[[0-9;]*[a-zA-Z]/g, '');
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)